packages/engine/src/gltf/GLTFComponent.tsx

```diff
@@ -40,17 +43,40 @@ import {
   useQuery,
   UUIDComponent
 } from '@etherealengine/ecs'
-import { dispatchAction, getState, useHookstate } from '@etherealengine/hyperflux'
+import { dispatchAction, getState, NO_PROXY, startReactor, useHookstate } from '@etherealengine/hyperflux'
 
 import { FileLoader } from '../assets/loaders/base/FileLoader'
 import { BINARY_EXTENSION_HEADER_MAGIC, EXTENSIONS, GLTFBinaryExtension } from '../assets/loaders/gltf/GLTFExtensions'
-import { ModelComponent } from '../scene/components/ModelComponent'
 import { SourceComponent } from '../scene/components/SourceComponent'
 import { SceneJsonType } from '../scene/types/SceneTypes'
 import { migrateSceneJSONToGLTF } from './convertJsonToGLTF'
 import { GLTFDocumentState, GLTFSnapshotAction } from './GLTFDocumentState'
 import { ResourcePendingComponent } from './ResourcePendingComponent'
 
+const loadDependencies = {
+  ['EE_model']: ['scene']
+} as Record<string, string[]>
+
+type ComponentDependencies = Record<ComponentType<typeof UUIDComponent>, Component[]>
+
+const buildComponentDependencies = (json: GLTF.IGLTF) => {
+  const dependencies = {} as ComponentDependencies
+  if (!json.nodes) return dependencies
+  for (const node of json.nodes) {
+    if (!node.extensions || !node.extensions[UUIDComponent.jsonID]) continue
+    const uuid = node.extensions[UUIDComponent.jsonID] as ComponentType<typeof UUIDComponent>
```

Comment:
`node.extensions[UUIDComponent.jsonID]` here is not of type `ComponentType<typeof UUIDComponent>` but just `EntityUUID`